src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -139,32 +177,38 @@ const NavHeader = (props) => {
           )}
         </View>
       </View>
-      <Animated.View style={[styles.headerContainer]}>
-        {!isHeaderHide ? (
-          <>
-            <Animated.View style={{opacity: opacityHeaderAnimation}}>
-              <FastImage
-                source={topicDetail?.icon_path ? {uri: topicDetail?.icon_path} : TopicDefaultIcon}
-                style={styles.image}
-              />
-            </Animated.View>
-            <View style={styles.containerAction}>
-              <Animated.View style={{opacity: opacityHeaderAnimation}}>
-                {isFollow ? (
-                  <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
-                ) : (
-                  <ButtonFollow handleSetFollow={handleFollowTopic} />
-                )}
-              </Animated.View>
-            </View>
-          </>
-        ) : null}
-      </Animated.View>
-      <View
-        onLayout={onDomainLayout}
-        style={[styles.domain(isHeaderHide), {bottom: getBottomPostition()}]}>
-        <TopicDomainHeader {...props} />
+      <View style={[styles.headerContainer]}>
+        <Animated.View style={{opacity: opacityHeaderAnimation}}>
+          <FastImage
+            source={
+              initialData.channelPicutre
+                ? {uri: initialData.channelPicutre}
+                : topicDetail?.icon_path
+                ? {uri: topicDetail?.icon_path}
+                : TopicDefaultIcon
+            }
+            style={styles.image}
+          />
+        </Animated.View>
+        <View style={styles.containerAction}>
+          <Animated.View style={{opacity: opacityHeaderAnimation}}>
+            {isLoading ? (
+              <Shimmer width={normalize(100)} height={normalize(36)} />
+            ) : isFollow ? (
+              <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
+            ) : (
+              <ButtonFollow handleSetFollow={handleFollowTopic} />
+            )}
+          </Animated.View>
+        </View>
       </View>
+      <Animated.View
+        style={[
+          styles.domain(isHeaderHide),
+          {bottom: normalize(20), opacity: opacityHeaderAnimation}
+        ]}>
+        <TopicDomainHeader {...props} />
+      </Animated.View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You've made changes to how the image source is determined in lines 183-188. You're now considering `initialData.channelPicutre` before falling back to `topicDetail?.icon_path` and then `TopicDefaultIcon`. This seems like a logical change, but make sure that `initialData.channelPicutre` always contains the most up-to-date and relevant image URL.



<!-- This is an auto-generated comment by OSS CodeRabbit -->